### ISSUE TITLE
[Bug]: Milvus will panic when backup data for v2.2.8 using latest backup tool 

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#24511]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.2.8
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka): rocksmq   
- SDK version(e.g. pymilvus v2.0.0rc2):v2.2.8
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2023/05/30 05:06:59.748 +00:00] [DEBUG] [rmq/rmq_producer.go:47] ["tr/send msg to stream"] [msg="send msg to stream done"] [duration=1.129149ms]
panic: implement me

goroutine 8442 [running]:
github.com/milvus-io/milvus/internal/distributed/proxy.(*Server).ListDatabases(0x1?, {0x1?, 0x3c2ee80?}, 0xc0041fefc0?)
        /go/src/github.com/milvus-io/milvus/internal/distributed/proxy/service.go:945 +0x27
github.com/milvus-io/milvus-proto/go-api/milvuspb._MilvusService_ListDatabases_Handler.func1({0x420afb8, 0xc0041ff290}, {0x3c2ee80?, 0xc0041fefc0})
        /go/pkg/mod/github.com/milvus-io/milvus-proto/go-api@v0.0.0-20230423035028-6708d3c12699/milvuspb/milvus.pb.go:11562 +0x7b
github.com/milvus-io/milvus/internal/proxy.RateLimitInterceptor.func1({0x420afb8, 0xc0041ff290}, {0x3c2ee80, 0xc0041fefc0}, 0xc0041d9ca0, 0xc0041dbd70)
        /go/src/github.com/milvus-io/milvus/internal/proxy/rate_limit_interceptor.go:38 +0xf0
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041ff290?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/milvus-io/milvus/internal/util/logutil.UnaryTraceLoggerInterceptor({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80, 0xc0041fefc0}, 0x422e5f8?, 0xc0041d9cc0)
        /go/src/github.com/milvus-io/milvus/internal/util/logutil/grpc_interceptor.go:22 +0x49
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/milvus-io/milvus/internal/proxy.UnaryServerInterceptor.func1({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80, 0xc0041fefc0}, 0x0?, 0xc0041d9ce0)
        /go/src/github.com/milvus-io/milvus/internal/proxy/privilege_interceptor.go:60 +0x5d
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/milvus-io/milvus/internal/proxy.UnaryServerHookInterceptor.func1({0x420afb8, 0xc0041ff1d0}, {0x3c2ee80, 0xc0041fefc0}, 0xc0041efe01?, 0xc0041d9d00)
        /go/src/github.com/milvus-io/milvus/internal/proxy/hook_interceptor.go:101 +0x962
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/auth.UnaryServerInterceptor.func1({0x420afb8, 0xc0041ff1d0}, {0x3c2ee80, 0xc0041fefc0}, 0xc0041d9ca0, 0xc0041d9d20)
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/auth/auth.go:47 +0xba
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041ff1d0?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1({0x420afb8, 0xc0041fef90}, {0x3c2ee80, 0xc0041fefc0}, 0xc0041d9ca0?, 0xc0041d9d40)
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38 +0x16a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x420afb8?, 0xc0041fef90?}, {0x3c2ee80?, 0xc0041fefc0?})
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x420afb8, 0xc0041fef90}, {0x3c2ee80, 0xc0041fefc0}, 0xc00258aaf0?, 0x3a548c0?)
        /go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf
github.com/milvus-io/milvus-proto/go-api/milvuspb._MilvusService_ListDatabases_Handler({0x3d8c040?, 0xc00047c2c0}, {0x420afb8, 0xc0041fef90}, 0xc0041f9260, 0xc000bba420)
        /go/pkg/mod/github.com/milvus-io/milvus-proto/go-api@v0.0.0-20230423035028-6708d3c12699/milvuspb/milvus.pb.go:11564 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000c2fc0, {0x421b850, 0xc0045a4000}, 0xc0041efc20, 0xc000bba600, 0x56e70c8, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0000c2fc0, {0x421b850, 0xc0045a4000}, 0xc0041efc20, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /go/pkg/mod/google.golang.org/grpc@v1.46.0/server.go:920 +0x28a
(milvus_backup) 

```

### Steps To Reproduce

```markdown
1. deploy standalone with v2.2.8
2. build milvus-backup using https://github.com/wayblink/milvus-backup/tree/db-dev
3. create backup
```

### Milvus Log

[standalone.log](https://github.com/milvus-io/milvus/files/11596213/standalone.log)

**Comment by xiaofan-luan** [comment]
seems to be a difficult to handle

**Comment by xiaofan-luan** [comment]
becasue 2.2.8 don't implement ListDatabases and it paniced

**Comment by yanliang567** [comment]
i think we can read the milvus version and handle it in different ways?
/assign @wayblink 
/unassign

**Comment by zhuwenxing** [comment]
> seems to be a difficult to handle

The problem is why Milvus panic rather than catches and throws the error when the API is not implemented.

Will there be the same issue (panic) for v2.2.9 when the server gets a request to call an API that is not implemented

**Comment by xiaofan-luan** [comment]
> > seems to be a difficult to handle
> 
> @xiaofan-luan The problem is why Milvus panic rather than catches and throws the error when the API is not implemented.
> 
> Will there be the same issue (panic) for v2.2.9 when the server gets a request to call an API that is not implemented

the trick is milvus 2.2.8 shouldn't panic on calling backup. but we can not fix the bug on 2.2.8 since it already released..

**Comment by zhuwenxing** [comment]
> the trick is milvus 2.2.8 shouldn't panic on calling backup. but we can not fix the bug on 2.2.8 since it already released..

So we need to fix this in the current version in case of next version SDK calls not implement API to v2.2.9 and make v2.2.9 panic.

**Comment by xiaofan-luan** [comment]
> > the trick is milvus 2.2.8 shouldn't panic on calling backup. but we can not fix the bug on 2.2.8 since it already released..
> 
> So we need to fix this in the current version in case of next version SDK calls not implement API to v2.2.9 and make v2.2.9 panic.

you can not fix it since 2.2.8 is already released...

**Comment by xiaofan-luan** [comment]
2.2.9 implemented the API so it won't panic
